cmd/swarmctl/node/common.go

```diff
@@ -138,6 +138,16 @@ func getNode(ctx context.Context, c api.ControlClient, input string) (*api.Node,
 		}
 
 		if l := len(rl.Nodes); l > 1 {
+			for _, n := range rl.Nodes {
+				name := n.Spec.Annotations.Name
```

Comment:
We don't seem to index nodes by the name in `Annotations.Name`. So it wouldn't be possible to look up a node by the `Annotations.Name` value without some API and store changes. This might be the right direction to go in, but I think it should be a separate PR and should be coordinated with changes to the Docker CLI. In the mean time, it doesn't make sense to me to display a name that can't be typed in to look up the node.